xrpl/utils/txn_parser/utils/balance_parser.py

```diff
@@ -0,0 +1,185 @@
+        assert isinstance(balance, dict)
+        balance_currency = balance.get("currency")
+        high_limit_issuer = high_limit.get("issuer")
+        if (
+            low_limit_issuer is not None
+            and balance_currency is not None
+            and high_limit_issuer is not None
+        ):
+            result = BalanceChange(
+                account=low_limit_issuer,
+                balance=Balance(
+                    currency=balance_currency,
+                    issuer=high_limit_issuer,
+                    value=f"{value.normalize():f}",
+                ),
+            )
+            return [result, _flip_trustline_perspective(result)]
+    return []
+
+
+def _group_balance_changes(
+    balance_changes: List[BalanceChange],
+) -> Dict[str, List[BalanceChange]]:
+    grouped_balance_changes: Dict[str, List[BalanceChange]] = {}
+    for change in balance_changes:
+        account = change["account"]
+        if account not in grouped_balance_changes:
+            grouped_balance_changes[account] = []
+        grouped_balance_changes[account].append(change)
+    return grouped_balance_changes
+
+
+def get_value(balance: Union[Balance, str]) -> Decimal:
+    """
+    Get a currency amount's value.
+
+    Args:
+        balance: Account's balance.
+
+    Returns:
+        The currency amount's value.
+    """
+    if isinstance(balance, str):
+        return Decimal(balance)
+    return Decimal(balance["value"])
+
+
+def get_quantities(
+    node: NormalizedNode,
+    value: Optional[Decimal],
+) -> List[BalanceChange]:
+    """
+    Retrieve the balance changes from a node.
+
+    Args:
+        node: The affected node.
+        value: The currency amount's value
+
+    Returns:
+        A list of balance changes.
+    """
+    if node["LedgerEntryType"] == "AccountRoot":
+        xrp_quantity = _get_xrp_quantity(node, value)
+        if xrp_quantity is None:
+            return []
+        return [xrp_quantity]
+    if node["LedgerEntryType"] == "RippleState":
+        trustline_quantity = _get_trustline_quantity(node, value)
+        return trustline_quantity
+    return []
+
+
+def group_by_account(
+    balance_changes: List[List[BalanceChange]],
+) -> List[BalanceChanges]:
+    """
+    Groups the balance changes in one list for each account.
+
+    Args:
+        balance_changes: All balance changes cause by a transaction.
+
+    Returns:
+        The grouped balance changes.
+    """
+    flattened_changes = list(chain.from_iterable(balance_changes))
+    grouped = _group_balance_changes(flattened_changes)
+    result = []
+    for account, account_balances in grouped.items():
+        balances: List[Balance] = []
+        for balance in account_balances:
+            balances.append(balance["balance"])
+        balance_changes_object = BalanceChanges(
+            account=account,
+            balances=balances,
+        )
+        result.append(balance_changes_object)
```

Comment:
```suggestion
        result.append(BalanceChanges(
            account=account,
            balances=balances,
        ))
```
Since this is a simple constructor, there's no need to create it in a separate line, plus the name `balance_changes_object` doesn't add much more info than just using the constructor.